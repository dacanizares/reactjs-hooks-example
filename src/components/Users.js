import React from 'react';
import { useState, useEffect } from 'react';
import User from './User';
import UsersApi from '../api/UsersApi';

function Users() {
  const [users, setUsers] = useState([]);
  const usersApi = new UsersApi();

  function fetchUsers() {
    usersApi.find().then(response => setUsers(response));
  }

  useEffect(() => {
    fetchUsers();
  }, []);

  return (
    <div>
      <div className='row'>
        {users.map((user, index) => <User user={user} key={index} />) }      
      </div>
      
      <div className='row mt-3'>
        <div className='col-xs-12'>
          <a href='#'
            className='btn btn-primary btn-lg' 
            onClick={() => fetchUsers()} >
            Update
          </a>
        </div>
      </div>
    </div>
  );
}

export default Users;