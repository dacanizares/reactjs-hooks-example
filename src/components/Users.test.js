import React from 'react';
import Users from './Users';
import renderer from 'react-test-renderer';

jest.mock('./Card', () => 'Card');

it('renders correctly', () => {
  const fecth = () => { }

  const userList = [{
    email: 'email@test.com', 
    name: {
      title: 'mr',
      first: 'name'
    },
    picture: {
      large: 'image here'
    }
  }];

  const tree = renderer
    .create(<Users usersList={userList} fetchUsers={fecth} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});