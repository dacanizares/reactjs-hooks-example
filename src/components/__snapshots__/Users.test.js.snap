// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="card-deck"
    >
      <Card
        imageUrl="image here"
        index={0}
        text="email@test.com"
        title="mr undefined"
      />
    </div>
  </div>
  <div
    className="row mt-3"
  >
    <div
      className="col-xs-12"
    >
      <a
        className="btn btn-primary btn-lg"
        href="#"
        onClick={[Function]}
      >
        Update
      </a>
    </div>
  </div>
</div>
`;
